llvm/docs/LangRef.rst

```diff
@@ -12880,7 +12880,10 @@ class <t_firstclass>` type.
    :ref:`fast-math flags <fastmath>`. These are optimization hints to enable
    otherwise unsafe floating-point optimizations. Fast-math flags are only valid
    for selects that return :ref:`supported floating-point types
-   <fastmath_return_types>`.
+   <fastmath_return_types>`. Note that the violation of poison-generating flags on
```

Comment:
The phrase "the violation of poison-generating flags" here is unclear.

```suggestion
   <fastmath_return_types>`. Note that the presence of value which would otherwise result in poison does not cause the result to be poison if the value is on the non-selected arm.
```